File: 1554. Strings Differ by One Character/main.js
/**
 * @param {string[]} dict
 * @return {boolean}
 */
let differByOne = function (dict) {
  let n = dict[0].length;
  let set = new Set();
  for (const str of dict) {
    for (let i = 0; i < n; i++) {
      let mask = str.substring(0, i) + "*" + str.substring(i + 1);
      if (set.has(mask)) return true;
      set.add(mask);
    }
  }
  return false;
};

File: 1554. Strings Differ by One Character/main.cpp
#include <iostream>
#include <unordered_map>
#include <unordered_set>
#include <vector>
using namespace std;

/**
 * I: dict = ["abcd","acbd", "aacd"]
 * O: 1
 *
 * I: dict = ["ab","cd","yz"]
 * O: 0
 *
 * I: dict = ["abcd","cccc","abyd","abab"]
 * O: 1
 */

bool differByOne(vector<string>& dict);

int main(int argc, char const* argv[]) {
    return 0;
}

// MLE
bool differByOne(vector<string>& dict) {
    int m = dict.size(), n = dict[0].size();
    unordered_set<string> set;
    for (int i = 0; i < m; i++) {
        for (int j = 0; j < n; j++) {
            string mask = dict[i].substr(0, j) + "#" + dict[i].substr(j + 1);
            if (set.find(mask) != set.end()) return true;
            set.insert(mask);
        }
    }
    return false;
}
